From e8bb168646d3fc5fc1c5bda6f8e775fe2c23aad4 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 28 Apr 2019 22:27:58 +0300
Subject: [PATCH] [SW-0] preboot: use APIs instead of include game

Instead of playing the include game in Makefile, use APIs for
receiving various device specific information and now be free to
include the device specific header files in device specific source
code files.
This stops the include game.

Change-Id: I2b7825def26d06f7d86e7d39e8ebe3570e54ab8a
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/Makefile b/Makefile
index 9472957..d500a49 100644
--- a/Makefile
+++ b/Makefile
@@ -14,7 +14,6 @@
 endif
 
 SFLAGS := $(CFLAGS)
-CFLAGS += -include $(HL_DEV)/$(HL_DEV).h
 
 LDFLAGS := -g -T common/preboot.lds -nostdlib -static
 LDFLAGS += -L ../../host/opt/ext-toolchain/lib/gcc/aarch64-linux-gnu/6.4.1
diff --git a/gaudi/gaudi.c b/gaudi/gaudi.c
index 61a4598..2884fbc 100644
--- a/gaudi/gaudi.c
+++ b/gaudi/gaudi.c
@@ -11,6 +11,7 @@
 #include "preboot.h"
 #include "pll.h"
 #include "uart.h"
+#include "gaudi.h"
 
 void preboot_pcie_wait_for_link(void)
 {
diff --git a/goya/goya.c b/goya/goya.c
index 5f14bae..3e9aa58 100644
--- a/goya/goya.c
+++ b/goya/goya.c
@@ -14,6 +14,7 @@
 
 #include "preboot.h"
 #include "pll.h"
+#include "goya.h"
 
 #define BTL_COUNTER_MAX		BIT(22)
 
